src/query/server/query.go

```diff
@@ -470,6 +472,22 @@ func Run(runOpts RunOptions) RunResult {
 		}
 		logger.Info("setup noop storage backend with etcd")
 
+	case config.DualStorageType:
+		logger.Info("setup dual storage backend")
+		opts, err := promremote.NewOptions(cfg.PrometheusRemoteBackend, scope, instrumentOptions.Logger())
+		if err != nil {
+			logger.Fatal("invalid configuration", zap.Error(err))
+		}
+		promRemoteStorage, err = promremote.NewStorage(opts)
+		if err != nil {
+			logger.Fatal("unable to setup prom remote backend", zap.Error(err))
+		}
+		defer func() {
+			if err := promRemoteStorage.Close(); err != nil {
+				logger.Error("error when closing storage", zap.Error(err))
+			}
+		}()
+		fallthrough
 	// Empty backend defaults to M3DB.
 	case "", config.M3DBStorageType:
```

Comment:
Interesting. I didn't know `fallthrough` leads to next case without matching the condition at all.
I though this case needs to be `case "", config.M3DBStorageType, config.DualStorageType:`, which seems more intuitive and readable to me.

Reply:
correct, the syntax was invalid, actually it kinda makes sense, in normal c++ code, we do the same:
```
switch(status): {
  case A:
      break;
  case B:
     // no break so B falls through C
  case C:
    // do sth
}
```